test/models/detector/ml_citation_test.rb

```diff
@@ -0,0 +1,162 @@
+        url = ENV.fetch('DETECTOR_LAMBDA_URL', nil)
+
+        assert(path && secret && url)
+        assert_predicate(Detector::MlCitation, :expected_env?)
+      end
+    end
+
+    test 'expected_env? returns false without DETECTOR_LAMBDA_CHALLENGE_SECRET set' do
+      with_enabled_mlcitation do
+        assert_predicate(Detector::MlCitation, :expected_env?)
+        ClimateControl.modify DETECTOR_LAMBDA_CHALLENGE_SECRET: nil do
+          assert_equal false, Detector::MlCitation.expected_env?
+        end
+      end
+    end
+
+    test 'expected_env? returns false without DETECTOR_LAMBDA_PATH set' do
+      with_enabled_mlcitation do
+        assert_predicate(Detector::MlCitation, :expected_env?)
+        ClimateControl.modify DETECTOR_LAMBDA_PATH: nil do
+          assert_equal false, Detector::MlCitation.expected_env?
+        end
+      end
+    end
+
+    test 'expected_env? returns false without DETECTOR_LAMBDA_URL set' do
+      with_enabled_mlcitation do
+        assert_predicate(Detector::MlCitation, :expected_env?)
+        ClimateControl.modify DETECTOR_LAMBDA_URL: nil do
+          assert_equal false, Detector::MlCitation.expected_env?
+        end
+      end
+    end
+
+    # These next three tests may not be needed, since the expected_env? method is tested above.
+    test 'DETECTOR_LAMBDA_CHALLENGE_SECRET is required' do
+      with_enabled_mlcitation do
+        ClimateControl.modify DETECTOR_LAMBDA_CHALLENGE_SECRET: nil do
+          result = Detector::MlCitation.new('ping')
+
+          assert_instance_of Detector::MlCitation, result
+
+          assert_nil(result.detections)
+        end
+      end
+    end
+
+    test 'DETECTOR_LAMBDA_PATH is required' do
+      with_enabled_mlcitation do
+        ClimateControl.modify DETECTOR_LAMBDA_PATH: nil do
+          result = Detector::MlCitation.new('ping')
+
+          assert_instance_of Detector::MlCitation, result
+
+          assert_nil(result.detections)
+        end
+      end
+    end
+
+    test 'DETECTOR_LAMBDA_URL is required' do
+      with_enabled_mlcitation do
+        ClimateControl.modify DETECTOR_LAMBDA_URL: nil do
+          result = Detector::MlCitation.new('ping')
+
+          assert_instance_of Detector::MlCitation, result
+
+          assert_nil(result.detections)
+        end
+      end
+    end
+    # End of the maybe-not-needed tests
+
+    # Class initalization
+    test 'lookup returns true when lambda running' do
+      with_enabled_mlcitation do
+        # These cassettes should be regenerated once the lambda is running in AWS. For now it will need to be running
+        # on localhost should the cassettes need to be regenerated.
+        VCR.use_cassette('lambda running') do
+          prediction = Detector::MlCitation.new('ping')
+
+          assert_instance_of Detector::MlCitation, prediction
+
+          assert(prediction.detections)
+        end
+      end
+    end
+
+    test 'lookup returns nil when challenge_secret is wrong' do
```

Comment:
This test is doing a lot of work by handling "all" of the possible error states. I'm not suggesting we test _every_ possible error that could come back, but I think a test that has a name that is clear it is testing non-200 status would be great.

```ruby
test 'non 200 http status responses result in no detection' do
...
end
```